pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -991,6 +991,12 @@ func (a *ApplicationBackupController) backupResources(
 	var err error
 	var resourceTypes []metav1.APIResource
 	// Listing all resource types
+	if len(backup.Spec.ResourceTypes) != 0 && len(backup.Spec.IncludeResources) != 0 {
```

Comment:
Also how does the case where the resourceType mentioned as PVCs is handled currently? Looks we are not going though ResourceType in the backupVolume().

Reply:
So basically we are looking at the use case if a user doesn't specify PVC in ResourceType do we still need to backupvolumes?  Need to discuss this use case. @adityadani / @ram-infrac ?

Reply:
Also the case, where do we need to backup all the volume backup in a namespace, if namespace is not present in the includeresources? Same criteria that we apply for other resources in the backupResources.

Reply:
Ideally I want to follow the same protocol for both resources and volumes, which means we need to change the `backupVolumes()` implementation, but we will break backward compatibility. How about we do this?

1. If a PVC resource is provided in `IncludeResources` then only include those specific PVCs in that namespace while backing up and ignore other PVCs from that namespace.
2. If PVC is provided as a type in the list of `ResourceTypes` then across all the namespaces backup only the volumes.
3. If `ResourceTypes` list is **NOT** empty, but it does **NOT** contain PVCs, then lets **NOT** backup PVCs -> This is not the current behavior but I think this will be the right direction since it is less confusing when someone is using this CRD. This will break backward compatibility but PX-Backup does not use this specific behavior, so the impact will be less.

Reply:
Yes, this makes sense and document the changes.